docker/docker/src/main/java/org/arquillian/cube/docker/impl/await/DockerHealthAwaitStrategy.java

```diff
@@ -0,0 +1,107 @@
+package org.arquillian.cube.docker.impl.await;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.InspectExecResponse;
+import com.github.dockerjava.api.exception.NotFoundException;
+import com.github.dockerjava.core.command.ExecStartResultCallback;
+import java.io.ByteArrayOutputStream;
+import java.util.logging.Logger;
+import org.arquillian.cube.docker.impl.client.config.Await;
+import org.arquillian.cube.docker.impl.docker.DockerClientExecutor;
+import org.arquillian.cube.docker.impl.util.Ping;
+import org.arquillian.cube.docker.impl.util.PingCommand;
+import org.arquillian.cube.spi.Cube;
+
+public class DockerHealthAwaitStrategy extends SleepingAwaitStrategyBase {
+
+    protected static final Logger log = Logger.getLogger(DockerHealthAwaitStrategy.class.getName());
+
+    public static final String TAG = "docker_health";
+
+    private static final int DEFAULT_POLL_ITERATIONS = 10;
+
+    private int pollIterations = DEFAULT_POLL_ITERATIONS;
+
+    private Cube<?> cube;
+    private DockerClient client;
+    private String[] command;
+
+    public DockerHealthAwaitStrategy(Cube<?> cube, DockerClientExecutor dockerClientExecutor, Await params) {
+        super(params.getSleepPollingTime());
+        this.cube = cube;
+        this.client = dockerClientExecutor.getDockerClient();
+
+        if (params.getIterations() != null) {
```

Comment:
How about moving this logic inside contrustor to one method and using it.

```java
    private int getIterations(Await params) {
        if (params.getIterations() != null) {
            return params.getIterations();
        } else {
            return DEFAULT_POLL_ITERATIONS;
        }
    }
```

Reply:
Can you check this last comment?

Reply:
sorry missed it
done